code/game/turfs/open/floor/light_floor.dm

```diff
@@ -52,21 +70,18 @@
 	set_light(0)
 	return ..()
 
-/turf/open/floor/light/attack_hand(mob/user)
-	. = ..()
-	if(.)
-		return
-	if(!can_modify_colour)
-		return
-	if(!on)
-		on = TRUE
-		currentcolor = 1
-		return
-	else
-		currentcolor++
-	if(currentcolor > coloredlights.len)
-		on = FALSE
-	update_icon()
+/turf/open/floor/light/attackby(obj/item/W, mob/user, params)
+	if(W.tool_behaviour == TOOL_MULTITOOL)
+		. = ..()
+		if(.)
+			return
+		if(!can_modify_colour)
+			return
+		var/choice = show_radial_menu(user,src, lighttile_designs, custom_check = FALSE, radius = 36, require_near = TRUE)
+		if(!choice)
+			return FALSE
+		currentcolor = choice
+		update_icon()
```

Comment:
Put this in `/turf/open/floor/light/multitool_act(mob/living/user)`

Reply:
Done!